src/test/java/io/confluent/idesidecar/websocket/messages/MessageSerializationTest.java

```diff
@@ -0,0 +1,61 @@
+package io.confluent.idesidecar.websocket.messages;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+public class MessageSerializationTest {
+
+  /** Test deserializing a simple message of an unknown message type, as if came from sidecar. */
+  @Test
+  public void testDeserializeRandomSidecarMessage() throws JsonProcessingException {
```

Comment:
We have a [utility](https://github.com/confluentinc/ide-sidecar/blob/9203449cce9f38a236f2c473d314b01ed6017802/src/test/java/io/confluent/idesidecar/restapi/util/ResourceIOUtil.java#L183) that will serialize and object, deserialize from that JSON, and test the objects are equal. There is another if we want to read a file from the test resources and then test serialization and deserialization.

See also https://github.com/confluentinc/ide-sidecar/blob/main/src/test/java/io/confluent/idesidecar/restapi/models/ConnectionsListTest.java for an example. It would be good to have multiple resource files with different permutations of valid and invalid representations.

Reply:
New test `testSerdeResourceMessageFiles()` written, proving sanity ser/deser round trips for each. Had to override equals() for `DynamicMessageBody` due to mutable map component so that equivalent maps will compare equals, otherwise `deserializeAndSerialize()` will fail for that message type. That took a while to narrow down.